### PR TITLE
docs: example for keyboard shortcut using custom js

### PR DESCRIPTION
**Pull request by thiswillbeyourgithub**
Signed-off-by: thiswillbeyourgithub <26625900+thiswillbeyourgithub@users.noreply.github.com>

## Description

Adds an example to show how to use `js` in the `head` argument of `Blocks` to add keyboard shortcut for any component. As suggested by @abidlabs in [a comment](https://github.com/gradio-app/gradio/issues/6101#issuecomment-1875565766).

Closes: #6101 #2670 #5223

## 🎯 PRs Should Target Issues

Before your create a PR, please check to see if there is [an existing issue](https://github.com/gradio-app/gradio/issues) for this change. If not, please create an issue before you create this PR, unless the fix is very small. 

Not adhering to this guideline will result in the PR being closed. 

## Tests

1. PRs will only be merged if tests pass on CI. To run the tests locally, please set up [your Gradio environment locally](https://github.com/gradio-app/gradio/blob/main/CONTRIBUTING.md) and run the tests: `bash scripts/run_all_tests.sh`

2. You may need to run the linters: `bash scripts/format_backend.sh` and `bash scripts/format_frontend.sh`

### CODE REVIEW

**Comment by thiswillbeyourgithub** [comment]
I think it deserves a mention in the `CHANGELOG.md` but I wasn't sure if i was supposed to do it myself or let the maintainers do it when releasing a version.

Here's the missing line just in case:
```
- [#6952](https://github.com/gradio-app/gradio/pull/6952) [`63c74e6`](https://github.com/gradio-app/gradio/pull/6952/commits/63c74e6f88b863fc5f630b5896e1d2ec11484633) - Show how to use JS to add keyboard shortcuts to any component. Thanks [@thiswillbeyourgithub](https://github.com/thiswillbeyourgithub/)!
```

**Comment by abidlabs** [comment]
Thanks @thiswillbeyourgithub! This looks good to merge, once we merge in @hannahblair's suggested changes.

Note that the changelog only reflects changes in the gradio library, not documentation, but you'll be included in GitHub's list of contributors!

File: guides/03_building-with-blocks/04_custom-CSS-and-JS.md [hunk 1]
```diff
@@ -87,4 +87,31 @@ with gr.Blocks(head=head) as demo:
...
+            document.getElementById("my_btn").click();
+        }
+    }
+}
+document.addEventListener('keypress', shortcuts, false);
+</script>
+"""
+
+with gr.Blocks(head=shortcut_js):
+    action_button = gr.Button(value="Name", elem_id="my_btn")
```
[the remark this answers is not in the document]
Comment (abidlabs) [reply guides/03_building-with-blocks/04_custom-CSS-and-JS.md:116]: Expanded this to include the full code for the demo (to make it easier for users to try out)

Also added `.toLowerCase()` (needed to work for me)

**Comment by abidlabs** [comment]
Merging. Thanks again @thiswillbeyourgithub!

**Pull request merged**